skimage/util/_flood_fill_cy.pyx

```diff
@@ -0,0 +1,212 @@
+
+    # Queue start position after clearing the buffer
+    queue_clear(queue_ptr)
+    queue_push(queue_ptr, &start_index)
+    flags[start_index] = FILL
+
+    # Break loop if all queued positions were evaluated
+    while queue_pop(queue_ptr, &current_index):
+        # Look at all neighboring samples
+        for i in range(neighbor_offsets.shape[0]):
+            neighbor = current_index + neighbor_offsets[i]
+
+            # Indexing sanity check
+            if neighbor < 0:
+                continue
+            if neighbor >= image.shape[0]:
+                continue
+
+            # Shortcut if neighbor is already part of fill
+            if flags[neighbor] == UNKNOWN:
+                if image[neighbor] == seed_value:
+                # Neighbor is part of fill - but must check its neighbors too.
+                    flags[neighbor] = FILL
+                    queue_push(queue_ptr, &neighbor)
+                else:
+                    # Do not check this point again
+                    flags[neighbor] = NOT_FILL
+
+
+cdef inline void _flood_fill_do_tol(
+        dtype_t[::1] image, unsigned char[::1] flags,
+        Py_ssize_t[::1] neighbor_offsets, QueueWithHistory* queue_ptr,
+        Py_ssize_t start_index, dtype_t seed_value, dtype_t high_tol,
+        dtype_t low_tol) nogil:
+    """Fill connected areas with 1, within a tolerance.
+
+    Parameters
+    ----------
+    image : ndarray, one-dimensional
+        The raveled view of a n-dimensional array.
+    flags : ndarray, one-dimensional
+        An array of flags that is used to store the state of each pixel during
+        evaluation.
+    neighbor_offsets : ndarray
+        A one-dimensional array that contains the offsets to find the
+        connected neighbors for any index in `image`.
+    queue_ptr :
+        Pointer to initialized queue.
+    start_index : int
+        Start position for the flood-fill.
+    seed_value :
+        Value of `image[start_index]`.
+    high_tol :
+        Upper limit for tolerance comparison.
+    low_tol :
+        Lower limit for tolerance comparison.
+    """
+    cdef:
+        QueueItem current_index, neighbor
+
+    # Queue start position after clearing the buffer
+    queue_clear(queue_ptr)
+    queue_push(queue_ptr, &start_index)
+    flags[start_index] = FILL
+
+    # Break loop if all queued positions were evaluated
+    while queue_pop(queue_ptr, &current_index):
+        # Look at all neighboring samples
+        for i in range(neighbor_offsets.shape[0]):
+            neighbor = current_index + neighbor_offsets[i]
+
+            # Indexing sanity check
+            if neighbor < 0:
+                continue
+            if neighbor >= image.shape[0]:
+                continue
+
+            # Only do comparisons on points not (yet) part of fill
+            if flags[neighbor] == UNKNOWN:
+                if image[neighbor] <= high_tol:
+                    if image[neighbor] >= low_tol:
```

Comment:
Is there a reason you are using two nested if-else-statements which could easily be merged? Is this for readability? 

Actually I think that
```python
if low_tol <= image[neighbor] <= high_tol:
    # Neighbor is in fill; must check its neighbors too.
    flags[neighbor] = FILL
    queue_push(queue_ptr, &neighbor)
else:
    flags[neighbor] = NOT_FILL
```
is clearer.